homeassistant/components/sensor/solaredge.py

```diff
@@ -0,0 +1,161 @@
+"""
+Support for SolarEdge Monitoring API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.solaredge/
+"""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import CONF_MONITORED_VARIABLES
+from homeassistant.components.light import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util import dt as dt_util
+
+REQUIREMENTS = ['solaredge==0.0.2']
+
+DOMAIN = "solaredge"
+
+# Config for solaredge monitoring api requests.
+CONF_API_KEY = "api_key"
+CONF_SITE_ID = "site_id"
+
+DELAY_OK = 10
+DELAY_NOT_OK = 20
+
+# Supported sensor types:
+# Key: ['name', unit, icon]
+SENSOR_TYPES = {
+    'lifeTimeData': ["Lifetime energy", 'Wh', 'mdi:solar-power'],
+    'lastYearData': ["Energy this year", 'Wh', 'mdi:solar-power'],
+    'lastMonthData': ["Energy this month", 'Wh', 'mdi:solar-power'],
+    'lastDayData': ["Energy today", 'Wh', 'mdi:solar-power'],
+    'currentPower': ["Current Power", 'W', 'mdi:solar-power']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SITE_ID): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=[]):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_entities,
+                         discovery_info=None):
+    """Create the SolarEdge Monitoring API sensor."""
+    api_key = config.get(CONF_API_KEY, None)
+    site_id = config.get(CONF_SITE_ID, None)
```

Comment:
`None` is not needed it's the default while using `dict.get()` and both are required and validated already. Thus we prefer it `dict[key]` is used.